src/hotspot/share/gc/shared/pretouchTask.cpp

```diff
@@ -64,13 +64,16 @@ void PretouchTask::work(uint worker_id) {
 
 void PretouchTask::pretouch(const char* task_name, char* start_address, char* end_address,
                             size_t page_size, WorkGang* pretouch_gang) {
-
+  // The chunk size should cap with the input page size which probably stands for
+  // large pages size if UseLargePages was set, otherwise processing chunks with
+  // much smaller size inside large size pages would hurt performance.
+  // Revising page_size should be placed after having decided the proper chuck_size.
```

Comment:
Something like

// Chunk size should be at least (unmodified) page size as using multiple threads pretouch on a single chunk can decrease performance

is sufficient here.

Reply:
Sure I will update this accordingly, thanks

Reply:
Done